File: src/AddTwoInts/add_client.py
import sys
import rospy
from beginning_with_ros.srv import AddTwoInts
from beginning_with_ros.srv import AddTwoIntsRequest
from beginning_with_ros.srv import AddTwoIntsResponse

def add_client(x, y):
    # Check if service exist
    rospy.wait_for_service('add_two_ints')
    try:
        # Initialize service provider
        serviceProxy = rospy.ServiceProxy('add_two_ints', AddTwoInts)

        # Get response
        response = serviceProxy(x, y)

        return response.sum

    except rospy.ServiceException(error):
        rospy.loginfo("Service call failed: {}".format(error))

if __name__ == "__main__":
    # Initialize node
    rospy.init_node("add_client_py")

    # Check argument number
    if len(sys.argv) == 3:
        x = int(sys.argv[1])
        y = int(sys.argv[2])
        response_sum = add_client(x, y)

        rospy.loginfo("")
        print("    Request: {}, {}".format(x, y))
        print("    Response from server: {}".format(response_sum))

    else:
        rospy.loginfo("Incorrect number of arguments - Need: x y")

File: src/Basics/AddTwoInts/add_client.cpp
#include "ros/ros.h"
#include "beginning_with_ros/AddTwoInts.h"
#include <stdio.h>
#include <stdlib.h>

int main (int argc, char **argv) {
    // initialize node
    ros::init(argc, argv, "add_client_cpp");
    ros::NodeHandle nodeHandle;

    // initialize client
    ros::ServiceClient client = nodeHandle.serviceClient<beginning_with_ros::AddTwoInts>("add_two_ints");
    
    // check argument number
    if (argc == 3) {
        beginning_with_ros::AddTwoInts srv;
        srv.request.a = atoll(argv[1]);
        srv.request.b = atoll(argv[2]);

        // check if client has called for service successfully
        // feed srv (with .request.a & .request.b, get back .response.sum)
        if (client.call(srv)) {
            ROS_INFO("");
            printf("    Request: %ld, %ld\n", (long int)srv.request.a, (long int)srv.request.b);
            printf("    Response from server: %ld\n", (long int)srv.response.sum);
        }

        else {
            ROS_ERROR("Service call failed");

            return 1;
        }
    }

    else {
        ROS_INFO("Incorrect number of arguments - Need: x y");

        return 1;
    }
    
    return 0;
}